### DISCUSSION TITLE
how to inference pointcloud stream.

### DISCUSSION THREAD

**Discussion by songhc8** [open-mmlab/mmdetection3d#2138] [Q&A] [answered]
Hi, 
Thanks for sharing this project. 

I would like to use this project under ROS2 environment. Could you provide one example about how to inference one pointcloud in numpy format without converting the numpy to bin file ?

for example, I subscribe one ros2 message _/pointcloud2_, the message can be convert to numpy format and stored in varibale "A", A is update in each cycle.  how to inference A. 

I have read pcd_demo.py, and pcd_seg_demo.py, both require one pcd file stored in .bin format. That means I should save variable A into .bin file first and read it into memory again?

Same question in Chinese:
你好，我想在ROS2下使用本项目，我看了项目的demo，都是给定一个.bin文件作为点云的输入。如果我是订阅的ros2消息，点云存在一个numpy变量里，怎么推理这个numpy变量呢？需要先存在硬盘生成.bin文件，再推理吗？是否有内存加载numpy变量推理的例程，谢谢。

**Comment by JingweiZhang12** [comment, marked as answer]
You can modify the loader interface of point clond in https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/loading.py#L403-L406
